query_optimizer/ExecutionGenerator.cpp

```diff
@@ -760,16 +760,59 @@ void ExecutionGenerator::convertCreateIndex(
   // CreateIndex is converted to a CreateIndex operator.
   const CatalogRelationInfo *input_relation_info =
       findRelationInfoOutputByPhysical(physical_plan->input());
+  // NOTE: the object pointed to by input_relation is owned by CatalogRelation
   CatalogRelation *input_relation =
       optimizer_context_->catalog_database()->getRelationByIdMutable(
             input_relation_info->relation->getID());
+
+  // check if any index with the specified name already exists
```

Comment:
Generally comments should be sentences (Sentences have a capitalized first word and a period following.)